torch/lib/c10d/ProcessGroupGloo.cpp

```diff
@@ -1896,11 +1896,17 @@ std::shared_ptr<ProcessGroup::Work> ProcessGroupGloo::gather(
   assertDense(invalidArgument, inputs);
 
   if (getRank() == opts.rootRank) {
-    if (outputs.size() != 1 ||
-        outputs[0].size() != static_cast<size_t>(getSize())) {
-      invalidArgument(
-          "requires a single-element output list "
-          "containing a list with <size> tensors");
+    if (outputs.size() != 1) {
+      std::stringstream ss;
+      ss << "requires a single-element output list containing a list with "
+         << getSize() << " tensors.";
+      invalidArgument(ss.str());
+    } else if (outputs[0].size() != static_cast<size_t>(getSize())) {
```

Comment:
For the user calling this code from the python api, this is the error condition that could happen. Internally in `distributed_c10d.py` we wrap the input into another list of size 1 (but the user is not exposed to that), and the error for that is handled above.